### DISCUSSION TITLE
Version 0.1 - Hello World!

### DISCUSSION THREAD

**Discussion by BenediktMKuehne** [e-m-b-a/embark#177] [Announcements]
<p align="center">
<br>
<a href="https://github.com/e-m-b-a/embark"><b>The first official EMBArk release is out now!</b></a>
<br><br>
<img src=https://github.com/e-m-b-a/emba/assets/497520/cf4305e9-290a-47d5-ba88-b863a07e572e width="700">
</p>

*****
Everything started as an idea in the beginning of 2021. The idea was to build an enterprise ready open source firmware analysis environment on top of _EMBA_. This environment should allow every product security team as well as every penetration tester and security researcher to use professional firmware analysis to improve the security of IoT/OT/ICS ... (you name it) devices as easy as possible. This idea was mixed up to an AMOS research project, where a team of students built a first PoC of _EMBArk_. You can find the original project [here](https://github.com/amosproj/amos2021ss01-emba-service). From there on continuous work, improvement and testing was running more or less under the radar. 
And today ... _EMBArk_ is stable and ready for more!

*****
EMBArk is built for a team, it supports multiple user uploading and analyzing firmware simultaneously.

**Your hardware is the limit!**

Tracking a devices security over time - through different firmware versions - has never been this easy.
EMBArk centralizes all EMBA-reports in one place and let's you do great things with the results.

Easy to setup and even easier to use.
Putting [EMBA](https://github.com/e-m-b-a/emba)s capabilities in the hands of everyone.

-> **EMBArk v0.1**

<hr /><em>This discussion was created from the release <a href='https://github.com/e-m-b-a/embark/releases/tag/v0.1'>Version 0.1 - Hello World!</a>.</em>

**Comment by m-1-k-3** [comment]
Great to have EMBArk v0.1 out now! Good work @BenediktMKuehne

**Comment by spon94** [comment]
The emba directory is empty，which version should i select？

**Reply by BenediktMKuehne** [reply]
Did you use the installer.sh script?

**Reply by spon94** [reply]
Yes，i ran `sudo ./installer.sh -d`,and the output is 
```zsh
EMBArk Installer
=================================================================
WARNING: This script can harm your environment!


==> Imported 1 necessary files

Default installation of EMBArk

Install debian packages for EMBArk installation
Hit:1 http://archive.ubuntu.com/ubuntu jammy InRelease
Hit:2 http://security.ubuntu.com/ubuntu jammy-security InRelease
Hit:3 http://archive.ubuntu.com/ubuntu jammy-updates InRelease
Hit:4 http://archive.ubuntu.com/ubuntu jammy-backports InRelease
Reading package lists... Done

Installation of the firmware scanner EMBA on host
fatal: not a git repository (or any of the parent directories): .git
fatal: not a git repository (or any of the parent directories): .git
./installer.sh: line 144: ./installer.sh: No such file or directory
Could not install EMBA
./installer.sh: line 146: ./emba: No such file or directory

EMBA installation failed
```

**Reply by m-1-k-3** [reply]
Please open a dedicated issue for handling installation problems.

Thanks

**Comment by BenediktMKuehne** [comment]
### **⚠️ The installer.sh script currently only supports git repositories for the normal default installation⚠️** 

That's an oversight on my part, sorry for that.

To install from sources do the following:
1. `sudo ./installer.sh -sd`
2. install emba into the emba folder manually (recommend as a git repo via clone)
